core/sdk-core/src/main/java/software/amazon/awssdk/core/waiters/PollingStrategy.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.core.waiters;
+
+import software.amazon.awssdk.annotations.SdkPublicApi;
+import software.amazon.awssdk.core.retry.backoff.BackoffStrategy;
+import software.amazon.awssdk.utils.Validate;
+
+/**
+ * Define the polling strategy for a {@link Waiter} to poll the resource
+ */
+@SdkPublicApi
+public final class PollingStrategy {
+
+    private final int maxAttempts;
+    private final BackoffStrategy backoffStrategy;
```

Comment:
I feel like a lot of customers would want "maxWaitTime" as a configuration more than maxAttempts. I realize the max wait time can be indirectly managed via the maxAttempts, backoffStrategy and client configuration, but it's not easy to do. It would be nice if we did it for them, and we can use the request-level max-execution-time and some checks before sleeping/delegating to the scheduled executor service to do so fairly easily.

Reply:
Is there any way to let the customer configure the `maxWaitTime` and `maxAttempts` on the generated service waiters, to override the behavior?

I don't mind waiting on customers to ask, but I have recently seen a *lot* of people using the `maxClientExecuteDuration` duration lately to limit the amount of time that the SDK will be holding their threads. I feel like SAs are recommending it pretty strongly to their customers, and I am worried that customers might think of the feature as incomplete without it.